package/network/config/ltq-vdsl-app/src/src/dsl_cpe_ubus.c

```diff
@@ -146,21 +151,23 @@ static inline void m_bool(const char *id, bool value) {
 }
 
 static inline void m_u32(const char *id, uint32_t value) {
-	blobmsg_add_u32(&b, id, value);
+	blobmsg_add_u64(&b, id, value);
 }
 
 static inline void m_str(const char *id, const char *value) {
 	blobmsg_add_string(&b, id, value);
 }
 
-static inline void m_db(const char *id, int value) {
-	m_double(id, (double)value / 10);
+static inline void m_db(const char *id, int value, int invalid) {
+	if (value != invalid) {
+		m_double(id, (double)value / 10);
+	}
 }
 
-static inline void m_array(const char *id, const uint8_t *value, uint8_t len) {
+static inline void m_array(const char *id, const uint8_t *value, uint16_t len) {
 	void *c = blobmsg_open_array(&b, id);
 
-	for (uint8_t i = 0; i < len; ++i)
+	for (uint16_t i = 0; i < len; ++i)
```

Comment:
When you already change it I would prefer `size_t` over `uint16_t` for both types. With size_t we should not run into an overflow.